src/leader_election.rs

```diff
@@ -170,7 +185,7 @@ pub mod ballot_leader_election {
         pub fn handle(&mut self, m: BLEMessage) {
             match m.msg {
                 HeartbeatMsg::Request(req) => self.handle_request(m.from, req),
-                HeartbeatMsg::Reply(rep) => self.handle_reply(rep),
+                HeartbeatMsg::Reply(rep) => self.handle_reply(m.from, rep),
```

Comment:
remove `m.from`. If we are using it only for logging, we can simply log the ballot in `rep`.